build/build.go

```diff
@@ -686,6 +690,14 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 						return err
 					}
 
+					record := metrics.Measure(ctx, "build.duration", "Measures the total build duration.",
```

Comment:
I think it doesn't make sense track this as "buildkit solve request" duration, per build ref. For buildx side, I would think the duration is for how long buildx command ran. What about all the other error returns that make buildx fail? There is a return in between these current start-end calls as well. We can discuss more, but I think we are interested in commands, not requests.

Reply:
I tried moving the location but it becomes difficult to get the parameters we need the further you move up the stack and it ends up relying on duplication between the `runControllerBuild` and `runBasicBuild` sections if I move it.

If you have a better location where this might fit, I can try to move it. I've now linked it with a progress writer because future metrics are going to read from the progress stream more directly so I think this location makes sense in that context.